### PR TITLE
Include `sf` in `sfdx`

### PR DESCRIPTION
**Pull request by mdonnalley**
### What does this PR do?

Includes `sf` in two ways:
- Adds `sf` to bin property in the package.json. This will handle npm installations
- Adds `scripts/include-sf.js` which creates the `sf` bin and cmd files. This script is run during the `pretarball` step

Depends on https://github.com/oclif/oclif/pull/617

### What issues does this PR fix or reference?

@W-9499638@

### CODE REVIEW

File: scripts/verify-tarballs [hunk 1]
```diff
@@ -131,11 +131,26 @@ function ensure_no_unexpected_files() {
     done
 }
 
+function ensure_sf_node_test() {
```
Comment (amphro) [inline scripts/verify-tarballs:134]: Should we actually try to run `sf --version` to make sure the file is working?

**Review by amphro — APPROVED** [review]
I added a DO NOT MERGE since this should be timed with the release of R1 and not earlier.

**Review by peternhale — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -65,22 +67,16 @@
...
+      "@oclif/plugin-command-snapshot",
+      "@salesforce/plugin-release-management"
     ],
     "update": {
       "s3": {
         "bucket": "dfc-data-production",
         "indexVersionLimit": 20,
         "folder": "media/salesforce-cli/sfdx",
         "acl": " ",
-        "host": "https://developer.salesforce.com",
```
Comment (mshanemc) [inline package.json:76]: where did all this functionality go?

Comment (mdonnalley) [reply package.json:76]: @mshanemc this configuration hasn't been needed since we moved to the new pack/upload/promote commands in oclif

**Review by amphro — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (mshanemc) [reply package.json:77]: I think we are using it 
https://github.com/oclif/oclif/search?q=s3.host

Comment (mshanemc) [reply package.json:77]: if that's true, could it be removed from here? https://github.com/oclif/oclif/blob/711132e6a29ba82e370a5c90cc5059682c389a0f/src/upload-util.ts#L23

Comment (mdonnalley) [reply package.json:77]: I believe there's a WI for that already

**Review by shetzel — APPROVED** [review]

**Pull request merged**